src/foundry_dev_tools/config/config.py

```diff
@@ -0,0 +1,189 @@
+
+    Profiles make configs like this possible:
+
+    .. code-block:: toml
+
+        [config]
+        example_option = 1
+
+        [integration.config]
+        example_option = 2
+
+    Where 'integration.config' will be loaded instead of 'config'
+    if the profile is set to integration.
+    """
+    config = _load_config_files(cfg_files())
+    config = merge_dicts(config, get_environment_variable_config())
+    if not config:
+        return None
+    if profile is None:
+        profile = config.get("profile")
+    if profile in ("config", "credentials"):
+        msg = f"Profile name can't be {profile}"
+        raise AttributeError(msg)
+    if profile is not None and (profile_config := config.get(profile)) and isinstance(profile_config, dict):
+        # merge the profile config with the non-prefixed config
+        config = {
+            "config": merge_dicts(config.get("config", {}), profile_config.get("config", {})),
+            "credentials": merge_dicts(config.get("credentials", {}), profile_config.get("credentials", {})),
+        }
+    return config
+
+
+def parse_credentials_config(config_dict: dict | None) -> TokenProvider:
+    """Parses the credentials config dictionary and returns a TokenProvider object."""
+    # check if there is a credentials config present
+    if config_dict is not None and (credentials_config := config_dict.get("credentials")):
+        # a domain must always be provided
+        if "domain" not in credentials_config:
+            raise MissingFoundryHostError
+
+        # create a host object with the domain and the optional scheme setting
+        host = Host(credentials_config.pop("domain"), credentials_config.pop("scheme", None))
+        # get the token provider config setting, if it does not exist use an empty dict
+        token_provider = credentials_config.pop("token_provider", {})
+        tp_name, tp_config = token_provider.get("name"), token_provider.get("config", {})
+
+        if tp_name:
+            if mapped_class := TOKEN_PROVIDER_MAPPING.get(tp_name):
+                # check the config kwargs and pass the valid kwargs to the mapped class
+                return mapped_class(**check_init(mapped_class, "credentials", {"host": host, **tp_config}))
+
+            # if the token_provider name was set but not present in the mapping
+            msg = f"The token provider implementation {tp_name} does not exist."
+            raise TokenProviderConfigError(msg)
+
+        # the jwt token provider needs a config
+        if tp_config:
+            ci = check_init(JWTTokenProvider, "credentials", {"host": host, **tp_config})
+            return JWTTokenProvider(**ci)
+        msg = (
+            "To authenticate with Foundry you need a TokenProvider. The token provider can be configured either via the"
+            " configuration file or the token_provider FoundryContext parameter."
+        )
+        raise TokenProviderConfigError(msg)
+    raise MissingCredentialsConfigError
```

Comment:
if no JWT, no auth, and APP_SERVICE_TS is set, use legacy streamlit/dash provider.